pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go

```diff
@@ -0,0 +1,218 @@
+// +kubebuilder:resource:categories=elastic,shortName=scp
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.readyCount",description="Resources configured"
+// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+type StackConfigPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StackConfigPolicySpec   `json:"spec,omitempty"`
+	Status StackConfigPolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// StackConfigPolicyList contains a list of StackConfigPolicy resources.
+type StackConfigPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StackConfigPolicy `json:"items"`
+}
+
+type StackConfigPolicySpec struct {
+	ResourceSelector metav1.LabelSelector          `json:"resourceSelector,omitempty"`
+	SecureSettings   []commonv1.SecretSource       `json:"secureSettings,omitempty"`
+	Elasticsearch    ElasticsearchConfigPolicySpec `json:"elasticsearch,omitempty"`
+}
+
+type ElasticsearchConfigPolicySpec struct {
+	// ClusterSettings holds the Elasticsearch cluster settings (/_cluster/settings)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ClusterSettings *commonv1.Config `json:"clusterSettings,omitempty"`
+	// SnapshotRepositories holds the Snapshot Repositories settings (/_snapshot)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	SnapshotRepositories *commonv1.Config `json:"snapshotRepositories,omitempty"`
+	// SnapshotLifecyclePolicies holds the Snapshot Lifecycle Policies settings (/_slm/policy)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	SnapshotLifecyclePolicies *commonv1.Config `json:"snapshotLifecyclePolicies,omitempty"`
+	// SecurityRoleMappings holds the Role Mappings settings (/_security/role_mapping)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// SecurityRoleMappings *commonv1.Config `json:"securityRoleMapping,omitempty"`
+	// AutoscalingPolicies holds the Autoscaling Policies settings (/_autoscaling/policy)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// AutoscalingPolicies *commonv1.Config `json:"autoscalingPolicies,omitempty"`
+	// IndexLifecyclePolicies holds the Index Lifecycle policies settings (/_ilm/policy)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// IndexLifecyclePolicies *commonv1.Config `json:"indexLifecyclePolicies,omitempty"`
+	// IngestPipelines holds the Ingest Pipelines settings (/_ingest/pipeline)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// IngestPipelines *commonv1.Config `json:"ingestPipelines,omitempty"`
+	// IndexTemplates holds the Index and Component Templates settings
+	// IndexTemplates *IndexTemplates `json:"indexTemplates,omitempty"`
```

Comment:
Are you going to enable these additional types?

Reply:
Yes, in a follow-up.